src/main/kotlin/no/nav/familie/ef/sak/repository/TilkjentYtelseRepository.kt

```diff
@@ -1,9 +1,46 @@
 package no.nav.familie.ef.sak.repository
 
+import no.nav.familie.ef.sak.repository.domain.Stønadstype
 import no.nav.familie.ef.sak.repository.domain.TilkjentYtelse
+import no.nav.familie.kontrakter.felles.oppdrag.OppdragIdForFagsystem
+import org.springframework.data.jdbc.repository.query.Query
+import java.time.LocalDate
 import java.util.*
 
 interface TilkjentYtelseRepository : RepositoryInterface<TilkjentYtelse, UUID>, InsertUpdateRepository<TilkjentYtelse> {
 
     fun findByPersonident(personident: String): TilkjentYtelse?
+
+    // language=PostgreSQL
+    @Query("""
+        SELECT ty.*
+            FROM tilkjent_ytelse ty
+                JOIN behandling b ON b.id = ty.behandling_id
+            WHERE b.fagsak_id = :fagsakId AND b.status = 'FERDIGSTILT'
+            ORDER BY b.opprettet_tid DESC
+            LIMIT 1""")
+    fun finnSisteTilkjentYtelse(fagsakId: UUID): TilkjentYtelse?
+
+    // language=PostgreSQL
+    @Query("""
+        SELECT behandling_id FROM (
+            SELECT b.id as behandling_id, row_number() over (PARTITION BY b.fagsak_id ORDER BY ty.opprettet_tid DESC) rn
+                FROM behandling b
+                    JOIN fagsak f on b.fagsak_id = f.id
+                    JOIN tilkjent_ytelse ty on b.id = ty.behandling_id -- att man har en tilkjent_ytelse
+                WHERE b.status = 'FERDIGSTILT' AND f.stonadstype = :stønadstype
+        ) q WHERE rn=1
```

Comment:
Mulig jeg ikke helt skjønner denne, men hvorfor kan vi ikke bare bruke enkel join og order_by som i spørringen over?

F.eks: 
```
Select b.id from behandling b
JOIN fagsak f on b.fagsak_id = f.id
JOIN tilkjent_ytelse ty on b.id=ty.behandling_id
where b.status='FERDIGSTILT' AND f.stonadstype = :stønadstype
order by ty.opprettet_tid DESC
LIMIT 1
```

Reply:
Du vill ha limit 1 før varje group

Reply:
Er inte resultatet av spørringen din, den "nyaste" behandlingen? Medan vi vill ha en nyaste behandlingen før varje fagsak

Reply:
Du har rett i at den henter nyeste for en stønadstype. 
Men det skal bare eksistere én fagsak per stønadstype (totalt 3 per person). Så usikker på forskjellen her?

Reply:
Vi ønsker å hente alle fagsaker sin siste behandling, gitt stønadstype

Reply:
Vi vil ha den nyaste behandlingen før varje fagsak som eksiterer , så i pseudokode
fagsakerList.foreach(fagsak -> finnSisteBehandlingen(fagsak))

Reply:
Åja, for alle personer 🙈

Reply:
(Kan jeg skylde på blåmandag? 😂)